### PR TITLE
Add file access check for file upload

### PR DESCRIPTION
**Pull request by abraunegg**
* Add a check to validate that we can read the file by reading the first 10MB when uploading a new file or resuming an upload

### CODE REVIEW

**Comment by abraunegg** [comment]
Fix validated by user in #113, awaiting review before merge

File: src/upload.d [hunk 1]
```diff
@@ -63,25 +63,36 @@ struct UploadSession
...
-			// request the session status
-			JSONValue response;
+			// Can we read the file - as a permissions issue or file corruption will cause a failure on resume
+			// https://github.com/abraunegg/onedrive/issues/113
 			try {
-				response = onedrive.requestUploadStatus(session["uploadUrl"].str);
-			} catch (OneDriveException e) {
-				if (e.httpStatusCode == 400) {
-					log.vlog("Upload session not found");
+				// attempt to read the first 10MB of the file
```
Comment (robertjf) [inline src/upload.d]: perhaps this test should be consolidated somewhere in a utility class to avoid duplication?

**Comment by abraunegg** [comment]
Essentially this patch adds only the following as a wrap around the original logic for a file upload - either as a simple file upload or a session based upload. The commit / change is just messy because of the shift in line spacing / indentations This is all that is really being added:

```
try {

	// attempt to read the first 10MB of the file

	read(path,10000000);
	
	... original code ...

} catch (std.file.FileException e) {

	// unable to read the new local file

	log.log("Skipping uploading this file as it cannot be read (file permissions or file corruption): ", path);

}
```

Whilst this code is duplicated in sync.d / upload.d to wrap around the original logic, if a class was made for the read() function, there would still be the need for the try{} catch {} logic for the function being called.

**Comment by robertjf** [comment]
or, you could put that entire block of code in a function returning a Boolean and catch the exception locally; reducing the need to wrap large amounts of code in a try-catch block?  Replace with an if statement instead...

**Pull request merged**